lib/Date/Utility.pm

```diff
@@ -1015,6 +1041,115 @@ sub today {
     return $today_obj;
 }
 
+=head2 _plus_years
+
+Returns a new Date::Utility object plus the given years. If the day is greater than days in the new month, it will take the day of end month.
+e.g.
+
+    print Date::Utility->new('2000-02-29')->_plus_years(1)->date_yyyymmdd;
+    # will got 2001-02-28
+
+=cut
+
+sub _plus_years {
+    my ($self, $years) = @_;
+    (looks_like_number($years) && $years == int($years)) || die "Need a integer years number";
+    return _create_trimmed_date($self, $self->year + $years, $self->month, $self->day_of_month);
```

Comment:
it looks you are loosing hour, minutes, seconds in that method. Are you sure, that this is required? 

I.e. I 'd expect that `->new("2018-02-04 10:45:22")->plus("1y")` will be `2019-02-04 10:45:22` and not `2019-02-04 00:00:00`.

Reply:
Please see the line https://github.com/binary-com/perl-Date-Utility/pull/23/files#diff-9982e44798acaa1521793b3f10130e02R1127 , the time part is in `$self` object.

Reply:
@basiliscos Maybe can you help me to give the function `_create_trimmed_date` a more meaningful name ?

Reply:
@chylli `_override($y, $m, $d, $h, $m, $s)` ?

Reply:
No, what I want to express is : `If the day is greater than the max day in that month , then use that max day as the day in the new object.`